Projects/UOContent/Mobiles/AI/BaseAI.cs

```diff
@@ -993,106 +899,131 @@ public virtual bool DoActionCombat()
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay("Praise the shepherd!");
+                m_Mobile.DebugSay("I am being herded.");
             }
 
             return true;
         }
-
+    
         var combatant = m_Mobile.Combatant;
-
-        if (combatant == null || combatant.Deleted || combatant.Map != m_Mobile.Map || !combatant.Alive ||
-            combatant.IsDeadBondedPet)
+    
+        if (!IsValidCombatant(combatant))
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay("My combatant is gone!");
+                m_Mobile.DebugSay("My combatant is missing.");
             }
 
-            Action = ActionType.Wander;
+            if (Action != ActionType.Wander)
+            {
+                Action = ActionType.Wander;
+            }
             return true;
         }
-
+    
         m_Mobile.Direction = m_Mobile.GetDirectionTo(combatant);
+    
         if (m_Mobile.TriggerAbility(MonsterAbilityTrigger.CombatAction, combatant))
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay($"I used my abilities on {combatant.Name}!");
+                m_Mobile.DebugSay(string.Format("I used my abilities on: {0}", combatant.Name));
             }
         }
-
+    
         return true;
     }
+    
+    private bool IsValidCombatant(Mobile combatant)
+    {
+        return combatant != null && !combatant.Deleted && combatant.Map == m_Mobile.Map
+            && combatant.Alive && !combatant.IsDeadBondedPet;
+    }
 
     public virtual bool DoActionGuard()
     {
-        if (Core.AOS && CheckHerding())
-        {
-            if (m_Mobile.Debug)
-            {
-                m_Mobile.DebugSay("Praise the shepherd!");
-            }
-        }
-        else if (Core.TickCount - m_NextStopGuard < 0)
+        if (Core.TickCount - m_NextStopGuard < 0)
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay("I am on guard");
+                m_Mobile.DebugSay("I am still on guard.");
             }
-            // m_Mobile.Turn( Utility.Random(0, 2) - 1 );
+
+            m_Mobile.Turn(Utility.Random(0, 2) - 1); // added for immersion
         }
         else
         {
             if (m_Mobile.Debug)
             {
-                m_Mobile.DebugSay("I stopped being on guard");
+                m_Mobile.DebugSay("I stopped being on guard.");
             }
 
-            Action = ActionType.Wander;
+            if (Action != ActionType.Wander)
```

Comment:
We don't need any of these action should not already equal what we want checks cause it is done in the setter.